estela-api/api/views/auth.py

```diff
@@ -206,26 +222,16 @@ def get_parameters(self, request):
         detail=False,
         permission_classes=[permissions.IsAuthenticated],
         authentication_classes=[TokenAuthentication],
+        serializer_class=ChangePasswordRequestSerializer,
     )
     def request(self, request, *args, **kwargs):
-        email = request.data["email"]
+        serializer = ChangePasswordRequestSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        email = serializer.validated_data['email']
         user = User.objects.filter(email=email)
         if not user:
             raise UserNotFoundError({"error": "User does not exist."})
         user = user.get()
-        if (
-            int(
-                (
-                    datetime.now(timezone.utc) - user.userprofile.last_password_change
-                ).total_seconds()
-            )
-            < settings.PASSWORD_CHANGE_TIME
-        ):
-            raise ChangePasswordError(
-                {
-                    "error": "You can only change your password every 6 months. Try again later."
-                }
-            )
         try:
             send_change_password_email(user, request)
         except Exception:
```

Comment:
The message in case of error should be `"...sending the password reset email."` rather than `"... sending the verification email."`.